src/DataStructures.jl

```diff
@@ -94,4 +94,7 @@ module DataStructures
     export status
     export deref_key, deref_value, deref, advance, regress
 
+    @deprecate iter Stack
+    @deprecate iter Queue
+
```

Comment:
This should do it:

```julia
@deprecate iter(s::Stack) backward_iter(s)
@deprecate iter(q::Queue) forward_iter(q)
```

Plenty of examples [here](https://github.com/JuliaLang/julia/blob/master/base/deprecated.jl#L104).